reference-adapter/pspcl-ifix-adapter/src/main/java/org/egov/repository/AuthTokenRepository.java

```diff
@@ -0,0 +1,64 @@
+package org.egov.repository;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.config.PspclIfixAdapterConfiguration;
+import org.egov.model.KeyCloackData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@Repository
+@Slf4j
+@NoArgsConstructor
+@AllArgsConstructor
+public class AuthTokenRepository {
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    PspclIfixAdapterConfiguration adapterConfiguration;
+
+    public static final String CLIENT_ID = "client_id";
+    public static final String CLIENT_SECRETE = "client_secret";
+    public static final String GRANT_TYPE = "grant_type";
+
+
+    public KeyCloackData getAuthToken() {
```

Comment:
When is this getting called? 
Auth token flow should be like:
1. Keep the token in redis or in-memory. 
2.  If the token is expired or unavailable, fetch and save the token. 
3. Reuse the same token for all subsequent requests.

Reply:
Intentionally, I haven't used the cache(Redis cache). Since pspcl-ifix-adapter con job is running once in a 24 hours . And I am using bulk fiscal events publish after all the reconciliation of all pspcl account number. hence , we need access_token only once per cron-job.  We can discuss as well.